pkg/wasm/proto/wsds.proto

```diff
@@ -0,0 +1,29 @@
+// Copyright Istio Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package istio.networking.wsds.v1;
+
+option go_package = "istio.io/istio/pkg/dns/proto/istio_networking_wsds_v1";
+
+// WasmSecret is a secret used to pull Wasm image.
+message WasmSecret {
+    // Name of the Wasm pull secret resource.
+    string name = 1;
+
+    // The secret that are used to pull Wasm images.
+    bytes secret = 2;
```

Comment:
I am curious the pros/cons compared to:
* Using Envoy standard SDS type
* Just putting it in the ECDS message (metadata?). Doesn't seem like there is any obvious metadata section but maybe there should be...

Reply:
> Using Envoy standard SDS type

Main problem with either SDS or separated resource is complexity and edge cases which is hard to cover. We need to coordinate ECDS update and SDS update (e.g. ECDS arrives at the agent, we read secret it needs, trigger a SDS request, then wait for secret to arrive etc). It will be even more complicated with gateway, since we need to intercept and mutate the request.

> Just putting it in the ECDS message (metadata?). Doesn't seem like there is any obvious metadata section but maybe there should be...

That is my preference, like you noticed, I did not find a good place to put it. We can potentially just use the VM env variables, but that requires unmarshal and marshal again at xds gen time, which would be very inefficient? I can explore more on this route if sending different resources are too hacky.

Reply:
> Main problem with either SDS or separated resource is complexity and edge cases which is hard to cover. 

Couldn't we still just put it in the same response message, like we are doing here

Reply:
> Couldn't we still just put it in the same response message, like we are doing here

Oh I see what you are saying. I thought you meant to use the existing SDS channel. Yeah, there is no particular reason to have a new type of resource, I can change it to Envoy Secret type.